// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 1 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
// start dotnet  restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c
// command took 1.83 sec and exited with 0
// start dotnet  build -c Release --no-restore /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1 /p:Deterministic=true /p:Optimize=true in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c
// command took 3.03 sec and exited with 0
// ***** Done, took 00:00:05 (5.02 sec)   *****
// Found 1 benchmarks:
//   ApiBenchmark.ExecuteApiRequest: DefaultJob

Setup power plan (GUID: 8c5e7fda-e8bf-4a96-9a85-a6e23a8c635c FriendlyName: Alto desempenho)
// **************************
// Benchmark: ApiBenchmark.ExecuteApiRequest: DefaultJob
// *** Execute ***
// Launch: 1 / 1
// Execute: dotnet 52111e5e-c103-4e48-9186-cbbd0ffb883c.dll --anonymousPipes 1560 196 --benchmarkName ApiBenchmark.ExecuteApiRequest --job Default --benchmarkId 0 in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c\bin\Release\netcoreapp3.1
// BeforeAnythingElse

// Benchmark Process Environment Information:
// BenchmarkDotNet v0.13.12
// Runtime=.NET Core 3.1.32 (CoreCLR 4.700.22.55902, CoreFX 4.700.22.56512), X64 RyuJIT AVX2
// GC=Concurrent Workstation
// HardwareIntrinsics=AVX2,AES,BMI1,BMI2,FMA,LZCNT,PCLMUL,POPCNT VectorSize=256
// Job: DefaultJob

OverheadJitting  1: 1 op, 287900.00 ns, 287.9000 us/op

System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.Net.Http.HttpRequestException: Nenhuma conexão pôde ser feita porque a máquina de destino as recusou ativamente.
 ---> System.Net.Sockets.SocketException (10061): Nenhuma conexão pôde ser feita porque a máquina de destino as recusou ativamente.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at ApiBenchmark.ExecuteApiRequest() in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\Domain\models\ApiBenchmark.cs:line 23
   at BenchmarkDotNet.Autogenerated.Runnable_0.<.ctor>b__3_3() in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c\52111e5e-c103-4e48-9186-cbbd0ffb883c.notcs:line 197
   at BenchmarkDotNet.Autogenerated.Runnable_0.WorkloadActionNoUnroll(Int64 invokeCount) in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c\52111e5e-c103-4e48-9186-cbbd0ffb883c.notcs:line 309
   at BenchmarkDotNet.Engines.Engine.RunIteration(IterationData data)
   at BenchmarkDotNet.Engines.EngineFactory.Jit(Engine engine, Int32 jitIndex, Int32 invokeCount, Int32 unrollFactor)
   at BenchmarkDotNet.Engines.EngineFactory.CreateReadyToRun(EngineParameters engineParameters)
   at BenchmarkDotNet.Autogenerated.Runnable_0.Run(IHost host, String benchmarkName) in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c\52111e5e-c103-4e48-9186-cbbd0ffb883c.notcs:line 176
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at BenchmarkDotNet.Autogenerated.UniqueProgramName.AfterAssemblyLoadingAttached(String[] args) in C:\Users\gusta\OneDrive\Documentos\Github\sync-async\benchmark\bin\Release\netcoreapp3.1\52111e5e-c103-4e48-9186-cbbd0ffb883c\52111e5e-c103-4e48-9186-cbbd0ffb883c.notcs:line 60
// AfterAll
No Workload Results were obtained from the run.
// Benchmark Process 24020 has exited with code -1.

// ** Remained 0 (0,0%) benchmark(s) to run. Estimated finish 2024-04-08 23:31 (0h 0m from now) **
Successfully reverted power plan (GUID: 381b4222-f694-41f0-9685-ff5bb260df2e FriendlyName: Equilibrado)
// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts\results\ApiBenchmark-report.csv
  BenchmarkDotNet.Artifacts\results\ApiBenchmark-report-github.md
  BenchmarkDotNet.Artifacts\results\ApiBenchmark-report.html

// * Detailed results *
ApiBenchmark.ExecuteApiRequest: DefaultJob
Runtime = .NET Core 3.1.32 (CoreCLR 4.700.22.55902, CoreFX 4.700.22.56512), X64 RyuJIT AVX2; GC = Concurrent Workstation
There are not any results runs

// * Summary *

BenchmarkDotNet v0.13.12, Windows 11 (10.0.22631.3296/23H2/2023Update/SunValley3)
11th Gen Intel Core i7-11390H 3.40GHz, 1 CPU, 8 logical and 4 physical cores
.NET SDK 8.0.200
  [Host]     : .NET Core 3.1.32 (CoreCLR 4.700.22.55902, CoreFX 4.700.22.56512), X64 RyuJIT AVX2
  DefaultJob : .NET Core 3.1.32 (CoreCLR 4.700.22.55902, CoreFX 4.700.22.56512), X64 RyuJIT AVX2


| Method            | Mean | Error |
|------------------ |-----:|------:|
| ExecuteApiRequest |   NA |    NA |

Benchmarks with issues:
  ApiBenchmark.ExecuteApiRequest: DefaultJob

// * Warnings *
Environment
  Summary -> Detected error exit code from one of the benchmarks. It might be caused by following antivirus software:
        - McAfee (\\?\C:\Program Files\McAfee\wps\1.14.212.1\mc-fw-host.exe)
        - Windows Defender (windowsdefender://)
Use InProcessEmitToolchain or InProcessNoEmitToolchain to avoid new process creation.


// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ns  : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
Run time: 00:00:06 (6.51 sec), executed benchmarks: 1

Global total time: 00:00:12 (12.38 sec), executed benchmarks: 1
// * Artifacts cleanup *
Artifacts cleanup is finished
